pkg/services/ngalert/sender/router_test.go

```diff
@@ -61,7 +61,7 @@ func TestIntegrationSendingToExternalAlertmanager(t *testing.T) {
 			"implementation":             "prometheus",
 		}),
 	}
-	alertsRouter := NewAlertsRouter(moa, fakeAdminConfigStore, mockedClock, appUrl, map[int64]struct{}{}, 10*time.Minute,
+	alertsRouter := NewAlertsRouter(moa.AlertmanagerFor, fakeAdminConfigStore, mockedClock, appUrl, map[int64]struct{}{}, 10*time.Minute,
```

Comment:
The only change in this file is passing a function instead of the struct to `NewAlertsRouter()`.